azurerm/internal/services/loganalytics/log_analytics_saved_search_resource.go

```diff
@@ -0,0 +1,222 @@
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_log_analytics_saved_search", *existing.ID)
+		}
+	}
+
+	parameters := operationalinsights.SavedSearch{
+		SavedSearchProperties: &operationalinsights.SavedSearchProperties{
+			Category:      utils.String(d.Get("category").(string)),
+			DisplayName:   utils.String(d.Get("display_name").(string)),
+			Query:         utils.String(d.Get("query").(string)),
+			FunctionAlias: utils.String(d.Get("function_alias").(string)),
+		},
+	}
+
+	if v, ok := d.GetOk("function_parameters"); ok {
+		attrs := v.(*schema.Set).List()
+		result := make([]string, 0)
+		for _, item := range attrs {
+			if item != nil {
+				result = append(result, item.(string))
+			}
+		}
+		parameters.SavedSearchProperties.FunctionParameters = utils.String(strings.Join(result, ", "))
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, name, parameters); err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, id.ResourceGroup, id.Name, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("cannot read Log Analytics Saved Search %q (WorkSpace %q / Resource Group %q): %s", name, id.Name, id.ResourceGroup, err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmLogAnalyticsSavedSearchRead(d, meta)
+}
+
+func resourceArmLogAnalyticsSavedSearchRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.SavedSearchesClient
+	workspaceClient := meta.(*clients.Client).LogAnalytics.WorkspacesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	id, err := parse.LogAnalyticsSavedSearchID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	workspace, err := workspaceClient.Get(ctx, id.ResourceGroup, id.WorkspaceName)
+	if err != nil {
+		if utils.ResponseWasNotFound(workspace.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf(" making Read request on AzureRM Log Analytics workspaces %q: %+v", id.Name, err)
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on AzureRM Log Analytics Saved Search %q (WorkSpace %q / Resource Group %q): %s", id.Name, id.WorkspaceName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("log_analytics_workspace_id", workspace.ID)
+
+	if props := resp.SavedSearchProperties; props != nil {
+		d.Set("display_name", props.DisplayName)
+		d.Set("category", props.Category)
+		d.Set("query", props.Query)
+		d.Set("function_alias", props.FunctionAlias)
+		if props.FunctionParameters != nil {
+			d.Set("function_parameters", strings.Split(*props.FunctionParameters, ", "))
+		}
```

Comment:
since this is a Set we need to ensure this is set in all circumstances; so can we make this:

```suggestion
		functionParams := make([]string, 0)
		if props.FunctionParameters != nil {
 			functionParams = strings.Split(*props.FunctionParameters, ", ")
 		}
		d.Set("function_parameters", functionParams)
```